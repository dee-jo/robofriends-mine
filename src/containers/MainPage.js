import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import './App.css';
import ErrorBoudary from '../components/ErrorBoundary';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter(robot => {
        return robot.name
          .toLowerCase()
          .includes(this.props.searchField.toLowerCase());
    });
  }

  render() {
    const { onSearchChange, robots, isPending } = this.props;
    const filteredRobots = this.filterRobots();

    return isPending? <h1>Loading</h1> :
      (
        <div className='tc'>
          <h1 className='f1'>Robo Friends</h1>
          <Scroll>
            <ErrorBoudary>
              <SearchBox searchChange={onSearchChange}></SearchBox>
              <CardList robots={ filteredRobots } /> 
            </ErrorBoudary>        
          </Scroll>
        </div>
      );
  }
}

export default MainPage;
