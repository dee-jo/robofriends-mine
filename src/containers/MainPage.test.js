import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
let wrapper2;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps} />);

  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'john',
    isPending: false
  }
  wrapper2 = shallow(<MainPage {...mockProps2} />);
});

test('Renders the MainPage without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Renders the MainPage without crashing with isPending=false', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: true
  }
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3).toMatchSnapshot();
});

test('Filters robots correctly for empty robots', () => {
  expect(wrapper.instance().filterRobots([])).toEqual([]);
})

test('Filters robots correctly for "john" search', () => {
  const robots = [{
    id: 3,
    name: 'John',
    email: 'john@gmail.com'
  }];
  
  expect(wrapper2.instance().filterRobots()).toEqual(robots);
})
 
test('Filters robots correctly for "john" search, tests for length', () => {
  
  expect(wrapper2.instance().filterRobots().length).toEqual(1);
})