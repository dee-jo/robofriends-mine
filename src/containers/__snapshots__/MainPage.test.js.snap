// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the MainPage without crashing 1`] = `ShallowWrapper {}`;

exports[`Renders the MainPage without crashing with isPending=false 1`] = `ShallowWrapper {}`;
